index.html

```diff
@@ -304,7 +304,8 @@ <h3>Normative Specifications</h3>
                     <dt id="hash" class="spec">RDF Dataset Hash (RDH)</dt>
                     <dd>
                         <p>
-                            This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These steps include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> using the algorithm specified in the “RDF Dataset Canonicalization” deliverable; (2) serializing the canonical form generated in the previous step to a byte stream in a deterministic way (e.g., by sorting a representation in <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>); and (3) application of a cryptographic hash function on the aforementioned deterministic byte stream. 
+                            This specification details the processing steps for calculating hash values of an arbitrary RDF Dataset. These steps include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> using the algorithm specified in the “RDF Dataset Canonicalization” deliverable; (2) serializing the canonical form generated in the previous step to a byte stream in some deterministic way (e.g., by sorting a representation in <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>). As a last step, a cryptographic hash function can be
```

Comment:
```suggestion
                            This specification details the processing steps for calculating hash values of an arbitrary RDF Dataset. These steps include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> using the algorithm specified in the “RDF Dataset Canonicalization” deliverable; (2) serializing the canonical form generated in the previous step to a byte stream in some deterministic way (e.g., by sorting a representation in <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>); and as a last step, (3) a cryptographic hash function can be
```
If this suggestion is not applied as written, then **_one_** of the following should be done:
* `deliverable; (2) serializing` should be changed to `deliverable; and (2) serializing` 
* `(1)` should be removed and `(2)` changed to `and`

Reply:
There's some additional discussion on this line over at #93 that I've recommended be folded into this PR's comment stream.

On that note, in #92 it was suggested that we want to give the WG the flexibility they need in these steps to produce something with appropriate security and performance characteristics -- and that the charter should only mention examples of how the serialization + hashing may occur, whilst making clear that we intend to make use of existing cryptographic hash algorithms for the actual hashing parts. I think the changes here (along with some minor fixes here from @TallTed's suggestions) are good enough in that respect.

As a side note on one of the serialization+hashing options raised by @aidhog in #92 I wanted to, for posterity, say that hashing quads independently and then XORing those hashes (to avoid having to sort) [is likely not a safe strategy](https://crypto.stackexchange.com/a/54546/8459). Regardless, the WG should make these decisions.

Reply:
In response to @TallTed's [comment](https://github.com/w3c/lds-wg-charter/pull/93#discussion_r770895124) on #92:

> Because including only one example format, and only including that when sorted, suggests that "sorted N-Quads" is preferred or even necessary.

@TallTed, sorting *is* necessary. It's not canonical if you don't do that. If my assertion were somehow to be wrong, however, using `e.g.,` language should avoid that. I think we're just trying to signal that there's already at least one solution to this problem that's being considered and that others may also be considered by the WG.

Reply:
I've closed #93 so we can fold the discussion in here as suggested (sorry for the noise). 

The latest version of the suggested text would be:

> `This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These step include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> for the RDF Dataset using the algorithm specified in the “RDF Dataset Canonicalization” deliverable, (2) application of a hash function over a serialization (such as sorted <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>, unsorted <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>, or <a href"https://www.w3.org/TR/trig/">TriG</a>) of the canonical form of the RDF Dataset.`

@dlongley, that's a useful reference about XORing, thanks! I guess it depends on whether or not we want to ensure cryptographic hashing, but for sure the proposal should not be incompatible with cryptographic concerns. So rather than XORing, something like the "MSet-Mu-Hash" function mentioned in that answer might be a good strategy. 

More generally, I think the WG should, however, aim for an [online algorithm](https://en.wikipedia.org/wiki/Online_algorithm) for RDH, as mentioned in the answer you linked. But this is work for the WG.

Reply:
@dlongley -- Fair points. That said, I think "sorted N-Quads" is equally non-canonical unless it includes some specification of what that sorting entails (I guess it must be lexical, and trim whitespace between `S,P,O,C`, but what then?  A->Z?  Z->A?  Unicode codepoint ASC/DESC?). So, to me, the "e.g., sorted N-Quads" must suggest a valid option, but it doesn't, as written.

Reply:
@TallTed, my goal was not to be overly prescriptive in the charter itself, but to mention, at a high level, that there's at least one option that would be fully described in a spec produced by the WG. Any "valid option" that we're going to use will need to be fully defined by the WG; there's nothing to reference yet other than elements of it (e.g., the N-Quads spec).

It seems your position is that we shouldn't mention any examples or inputs into that process that will be considered by the WG in this particular section of text. If you are ok with mentioning this sort of information -- what language would work for you? Could we say: `(e.g., N-Quads [link] canonicalized and sorted via an algorithm specified by the WG)`? Could we say `(e.g., see input documents for potential serializations)`?

Reply:
I changed the paragraph by taking the one of @aidhog and also simplifying it a bit. Here it goes now:

> This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These step include the generation of a <a href="./explainer.html#canonical_form">canonical form</a> for the RDF Dataset using the algorithm specified in the “RDF Dataset Canonicalization” deliverable, and the application of a hash function over a serialization (such as sorted or unsorted <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>) of the canonical form of the RDF Dataset.

I think this leaves things open enough for the charter. @aidhog I have removed the reference to TriG; not sure whether that is viable in view of the earlier remarks, and would mean a completely different approach I believe. It would just trigger unnecessary discussions at this point...